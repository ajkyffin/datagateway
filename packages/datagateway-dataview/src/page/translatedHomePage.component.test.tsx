import React from 'react';
import { createShallow } from '@material-ui/core/test-utils';
import {
  TranslatedHomePage as HomePage,
  TranslatedHomePageStateProps,
} from './translatedHomePage.component';

describe('HomePage', () => {
  let shallow;
  let props: TranslatedHomePageStateProps;

  beforeEach(() => {
    shallow = createShallow({ untilSelector: 'div' });

    props = {
      pluginHost: 'test',
    };
  });

  it('translated homepage renders correctly', () => {
    const wrapper = shallow(<HomePage {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
