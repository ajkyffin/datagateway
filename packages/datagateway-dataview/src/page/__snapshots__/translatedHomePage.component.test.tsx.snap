// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage translated homepage renders correctly 1`] = `
<div
  id="dg-homepage"
>
  <div
    className="makeStyles-backgroundImage-1"
  >
    <div
      className="makeStyles-backgroundDecals-2"
    >
      <Styled(MuiBox)
        style={
          Object {
            "left": "50%",
            "position": "relative",
            "top": "45px",
            "transform": "translate(-50%)",
          }
        }
      >
        <WithStyles(ForwardRef(Typography))
          className="makeStyles-backgroundTitle-3"
          variant="h2"
        >
          <Trans
            i18nKey="homePage.title_line1"
          >
            <strong>
              Data discovery
            </strong>
             and 
            <strong>
              access
            </strong>
          </Trans>
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))
          className="makeStyles-backgroundTitle-3"
          variant="h2"
        >
          <Trans
            i18nKey="homePage.title_line2"
          >
            for 
            <strong>
              large-scale
            </strong>
            science facilities
          </Trans>
        </WithStyles(ForwardRef(Typography))>
      </Styled(MuiBox)>
    </div>
  </div>
  <Styled(MuiBox)
    className="makeStyles-contentBox-4"
  >
    <WithStyles(ForwardRef(Paper))
      className="makeStyles-paper-5"
      elevation={1}
    >
      <WithStyles(ForwardRef(Grid))
        container={true}
        style={
          Object {
            "height": "100%",
          }
        }
      >
        <WithStyles(ForwardRef(Grid))
          item={true}
          xs={6}
        >
          <Styled(MuiBox)
            className="makeStyles-paperContent-7"
          >
            <WithStyles(ForwardRef(Typography))
              className="makeStyles-paperHeading-10"
              variant="h4"
            >
              homePage.browse.title
            </WithStyles(ForwardRef(Typography))>
            <WithStyles(ForwardRef(Typography))
              className="makeStyles-paperDescription-11"
              variant="body1"
            >
              homePage.browse.description1
            </WithStyles(ForwardRef(Typography))>
            <WithStyles(ForwardRef(Typography))
              className="makeStyles-paperDescription-11"
              variant="body1"
            >
              <Trans
                i18nKey="homePage.browse.description2"
              >
                <strong>
                  DataGateway
                </strong>
                 focuses on providing data discovery and data access functionality to the data.
              </Trans>
            </WithStyles(ForwardRef(Typography))>
            <Styled(MuiBox)
              marginTop="auto"
            >
              <WithStyles(ForwardRef(Button))
                aria-label="homePage.browse.button_arialabel"
                color="primary"
                href="homePage.browse.link"
                variant="contained"
              >
                homePage.browse.button
              </WithStyles(ForwardRef(Button))>
            </Styled(MuiBox)>
          </Styled(MuiBox)>
        </WithStyles(ForwardRef(Grid))>
        <WithStyles(ForwardRef(Grid))
          item={true}
          xs={6}
        >
          <div
            className="makeStyles-browseBackground-14"
          >
            <div
              className="makeStyles-browseDecal-15"
            />
          </div>
        </WithStyles(ForwardRef(Grid))>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Paper))>
    <WithStyles(ForwardRef(Grid))
      container={true}
      spacing={2}
    >
      <WithStyles(ForwardRef(Grid))
        item={true}
        sm={4}
        xs={12}
      >
        <WithStyles(ForwardRef(Paper))
          className="makeStyles-paper-5"
          elevation={1}
        >
          <Styled(MuiBox)
            className="makeStyles-paperContent-7"
          >
            <WithStyles(ForwardRef(Avatar))
              className="makeStyles-avatar-8"
            >
              <Memo(ForwardRef(SearchIcon))
                className="makeStyles-avatarIcon-9"
              />
            </WithStyles(ForwardRef(Avatar))>
            <WithStyles(ForwardRef(Typography))
              className="makeStyles-paperHeading-10"
              variant="h4"
            >
              homePage.search.title
            </WithStyles(ForwardRef(Typography))>
            <WithStyles(ForwardRef(Typography))
              className="makeStyles-paperDescription-11"
              variant="body1"
            >
              homePage.search.description
            </WithStyles(ForwardRef(Typography))>
            <Styled(MuiBox)
              marginTop="auto"
            >
              <WithStyles(ForwardRef(Button))
                aria-label="homePage.search.button_arialabel"
                color="primary"
                href="homePage.search.link"
                variant="contained"
              >
                homePage.search.button
              </WithStyles(ForwardRef(Button))>
            </Styled(MuiBox)>
          </Styled(MuiBox)>
        </WithStyles(ForwardRef(Paper))>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        sm={4}
        xs={12}
      >
        <WithStyles(ForwardRef(Paper))
          className="makeStyles-paper-5"
          elevation={1}
        >
          <Styled(MuiBox)
            className="makeStyles-paperContent-7"
          >
            <WithStyles(ForwardRef(Avatar))
              className="makeStyles-avatar-8"
            >
              <Memo(ForwardRef(GetAppIcon))
                className="makeStyles-avatarIcon-9"
              />
            </WithStyles(ForwardRef(Avatar))>
            <WithStyles(ForwardRef(Typography))
              className="makeStyles-paperHeading-10"
              variant="h4"
            >
              homePage.download.title
            </WithStyles(ForwardRef(Typography))>
            <WithStyles(ForwardRef(Typography))
              className="makeStyles-paperDescription-11"
              variant="body1"
            >
              homePage.download.description
            </WithStyles(ForwardRef(Typography))>
            <Styled(MuiBox)
              marginTop="auto"
            >
              <WithStyles(ForwardRef(Button))
                aria-label="homePage.download.button_arialabel"
                color="primary"
                href="homePage.download.link"
                variant="contained"
              >
                homePage.download.button
              </WithStyles(ForwardRef(Button))>
            </Styled(MuiBox)>
          </Styled(MuiBox)>
        </WithStyles(ForwardRef(Paper))>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        sm={4}
        xs={12}
      >
        <WithStyles(ForwardRef(Paper))
          className="makeStyles-bluePaper-6"
          elevation={1}
        >
          <div
            className="makeStyles-facilityDecal-16"
          >
            <Styled(MuiBox)
              className="makeStyles-paperContent-7"
            >
              <WithStyles(ForwardRef(Typography))
                className="makeStyles-bluePaperHeading-12"
                variant="h4"
              >
                homePage.facility.title
              </WithStyles(ForwardRef(Typography))>
              <WithStyles(ForwardRef(Typography))
                className="makeStyles-bluePaperDescription-13"
                variant="body1"
              >
                homePage.facility.description
              </WithStyles(ForwardRef(Typography))>
              <Styled(MuiBox)
                marginTop="auto"
              >
                <WithStyles(ForwardRef(Button))
                  aria-label="homePage.facility.button_arialabel"
                  className="makeStyles-lightBlueButton-17"
                  color="primary"
                  href="homePage.facility.link"
                  variant="contained"
                >
                  homePage.facility.button
                </WithStyles(ForwardRef(Button))>
              </Styled(MuiBox)>
            </Styled(MuiBox)>
          </div>
        </WithStyles(ForwardRef(Paper))>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
  </Styled(MuiBox)>
</div>
`;
